modin/pandas/groupby.py

```diff
@@ -128,7 +129,46 @@ def sem(self, ddof=1):
         return self._default_to_pandas(lambda df: df.sem(ddof=ddof))
 
     def mean(self, *args, **kwargs):
-        return self._apply_agg_function(lambda df: df.mean(*args, **kwargs))
+        fallback = False
+        converted_columns = {}
+        if not kwargs.get("numeric_only", False):
+            for col, dt in self._query_compiler.dtypes.items():
+                if is_datetime64_any_dtype(dt):
+                    if self._df[col].hasnans:
+                        fallback = True
+                        break
+                    else:
+                        converted_columns[col] = dt
+        if fallback:
```

Comment:
I think renaming into `groupby_mean_numeric` is the best approach. Otherwise it is not clear how to handle `mean` aggregation in query_compiler separately from all other kinds of aggregations.